tmt/steps/provision/minute.py

```diff
@@ -0,0 +1,250 @@
+    (see 1minutetip list and 1minutetip list-flavors).
+
+    """
+    # Guest instance
+    _guest = None
+
+    # Supported methods
+    _methods = [
+        tmt.steps.Method(name='1minutetip', doc=__doc__, order=50),
+        ]
+
+    @classmethod
+    def options(cls, how=None):
+        """ Prepare command line options for 1minutetip """
+        return [
+            click.option(
+                '-i', '--image', metavar='IMAGE',
+                help='Select image to use. See 1minutetip list for available choices.'),
+            click.option(
+                '--flavor', metavar='FLAVOR',
+                help='Select flavor to use. See 1minutetip list-flavors for available choices.'),
+            click.option(
+                '--openstack', metavar='URL', required=True,
+                help='URL of openstack API to use for the provision.'),
+            ] + super().options(how)
+
+    def _get_default_image(self):
+        _, images = run_openstack(self.get('openstack'),
+                'image list -f value -c Name', True)
+        # Use the latest Fedora image
+        fedora_re = re.compile(r'1MT-Fedora-(?P<ver>\d+)')
+        fedora_images = [image for image in images.splitlines() if fedora_re.match(image)]
+        return max(fedora_images, key=lambda x: int(fedora_re.match(x).group('ver')))
+    
+    def default(self, option, default=None):
+        """ Return the default value for the given option """
+        defaults = {
+            'image': self._get_default_image(),
+            'flavor': 'm1.small'
+        }
+        return defaults.get(option, default)
+
+    def show(self):
+        """ Show provision details """
+        super().show(['image', 'flavor'])
+
+    def wake(self, data=None):
+        """ Override options and wake up the guest """
+        super().wake(['image', 'flavor', 'openstack'])
+        if data:
+            self._guest = Guest1minutetip(data, name=self.name, parent=self.step)
+            self._guest.wake()
+
+    def go(self):
+        """ Provision the container """
+        super().go()
+
+        data = dict(user=DEFAULT_USER)
+        for opt in ['image', 'flavor', 'openstack']:
+            val = self.get(opt)
+            if opt != 'openstack':
+                self.info(opt, val, 'green')
+            data[opt] = val
+
+        self._guest = Guest1minutetip(data, name=self.name, parent=self.step)
+        self._guest.start()
+
+    def guest(self):
+        """ Return the provisioned guest """
+        return self._guest
+
+
+class Guest1minutetip(tmt.Guest):
+    """
+    1minutetip instance
+
+    The following keys are expected in the 'data' dictionary:
+        
+        image ...... 1minutetip image name
+        flavor ..... openstack server flavor to use
+        openstack .. URL of openstack API
+    """
+    def load(self, data):
+        super().load(data)
+        self.key = SSH_KEY
+        self.openstack = data.get('openstack')
```

Comment:
hmm, this is not really openstack .. this is 1minutetip's openstack api url ?

I believe naming the option `api-url` would make more sense